packages/web/app/views/patients/panes/SummaryPane.jsx

```diff
@@ -1,23 +1,69 @@
 import React, { useCallback, useState } from 'react';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
 import { usePatientNavigation } from '../../../utils/usePatientNavigation';
 import { useEncounter } from '../../../contexts/Encounter';
 import { Box } from '@material-ui/core';
 
-import { CompactContentPane as ContentPane } from '../../../components';
+import {
+  CompactContentPane as ContentPane,
+  LargeBodyText,
+  Modal,
+  ModalActionRow,
+  TranslatedText,
+} from '../../../components';
 import { PatientEncounterSummary } from '../components/PatientEncounterSummary';
 import { PatientHistory } from '../../../components/PatientHistory';
 import { EncounterModal } from '../../../components/EncounterModal';
 import { LocationBookingsTable } from '../../../components/Appointments/LocationBookingsTable';
 import { useAuth } from '../../../contexts/Auth';
 import { useSettings } from '../../../contexts/Settings';
 import { OutpatientAppointmentsTable } from '../../../components/Appointments/OutpatientAppointmentsTable';
+import { useApi } from '../../../api';
+import styled from 'styled-components';
+
+const StyledLargeBodyText = styled(LargeBodyText)`
+  margin: 20px;
+`;
+
+export const ExistingEncounterWarningModal = React.memo(({ open, onClose }) => {
+  return (
+    <Modal
+      title={
+        <TranslatedText
+          stringId="patient.modal.activeEncounterWarning.title"
+          fallback="Cannot create new encounter"
+        />
+      }
+      width="md"
+      open={open}
+      onClose={onClose}
+      data-testid="formmodal-4oua"
+    >
+      <StyledLargeBodyText>
+        <TranslatedText
+          stringId="patient.encounterSummary.activeEncounterWarning"
+          fallback="This patient has an active encounter. The active encounter must be discharged before a new encounter can be created."
+        />
+      </StyledLargeBodyText>
+      <ModalActionRow onConfirm={onClose} confirmText="Close" />
+    </Modal>
+  );
+});
 
 export const SummaryPane = React.memo(({ patient, additionalData, disabled }) => {
-  const [isModalOpen, setModalOpen] = useState(false);
+  const [isCheckInModalOpen, setIsCheckInModalOpen] = useState(false);
+  const [isWarningModalOpen, setIsWarningModalOpen] = useState(false);
   const { navigateToEncounter } = usePatientNavigation();
   const { loadEncounter } = useEncounter();
-  const { ability } = useAuth();
+  const { ability, facilityId } = useAuth();
   const { getSetting } = useSettings();
+  const queryClient = useQueryClient();
+  const api = useApi();
+  const { refetch: refetchCurrentEncounter } = useQuery(
+    ['patientCurrentEncounter', patient.id],
```

Comment:
I think u can do this like this too kinda clean
```suggestion
    ['patient', patient.id, 'currentEncounter'],
```

Reply:
Id have to change in a few places and feels a bit out of scope imo